api/src/core/Directus/Database/TableGateway/RelationalTableGateway.php

```diff
@@ -1040,23 +1049,10 @@ public function createMetadataPagination(array $metadata = [], array $params = [
 
         $meta_param = explode(',', ArrayUtils::get($params, 'meta', ''));
 
-        if ((in_array('filter_count', $meta_param) || in_array('*', $meta_param))) {
-            $metadata['filter_count'] = $countedData['total_count'];
-        }
-
-        if ($filtered) {
-            $filteredparams = array_merge($params, [
-                "depth" => 0,
-                "fields" => $this->primaryKeyFieldName,
-                "limit" => -1
-            ]);
-
-            $entries = $this->fetchItems($filteredparams);
-            $total = count($entries);
-            if (in_array('filter_count', $meta_param) || in_array('*', $meta_param)) {
-                $metadata['filter_count'] = $total;
-            }
+        if ($filtered && (in_array('filter_count', $meta_param) || in_array('*', $meta_param) || in_array('page', $meta_param))) {
```

Comment:
admin panel uses 'filter_count' param to display num of elements in collection, in your case, if there will be no filters, but 'filter_count' is defined in request query, it won't return it.
The second moment is that $total variable, declared above, becomes useless. I think it should be like this:

```php
if ($filtered) {
            $total = $this->countFilterTotal($params);
            $filtered_count = $total;
}
if ((in_array('filter_count', $meta_param) || in_array('*', $meta_param))) {
        $metadata['filter_count'] = isset($filtered_count) ? $filtered_count : $countedData['total_count'];
}
```